internal/infrastructure/bot/handlers.go

```diff
@@ -282,3 +287,52 @@ func (b *Bot) PrivateSummary(i *discordgo.InteractionCreate) error {
 	_, err = b.mgr.SessionForGuild(gID).FollowupMessageCreate(i.Interaction, false, &discordgo.WebhookParams{Content: "Check your DM!"})
 	return err
 }
+
+func (b *Bot) SetWorld(i *discordgo.InteractionCreate) error {
+	guildID := i.GuildID
+	userID := i.Member.User.ID
+
+	// Fetch guild to check owner
+	guild, err := b.mgr.Gateway.Guild(guildID)
+	if err != nil {
+		return fmt.Errorf("could not fetch guild: %w", err)
+	}
+	if guild.OwnerID != userID {
+		return fmt.Errorf("only the server owner can use this command")
+	}
+
+	world := ""
+	for _, opt := range i.ApplicationCommandData().Options {
+		if opt.Name == "world" {
+			world = opt.StringValue()
+		}
+	}
+	if world == "" {
+		return fmt.Errorf("world name is required")
+	}
+
+	// Save or update the world name for this guild in the database
+	err = b.SetGuildWorld(guildID, world)
+	if err != nil {
+		return fmt.Errorf("failed to save world name: %w", err)
+	}
+
+	// Configure the online checker with the new world name
+	b.ConfigureWorldNameForGuild(guildID)
+
+	gID, err := stringsHelper.StrToInt64(guildID)
+	if err != nil {
+		return fmt.Errorf("could not parse guild id: %v", guildID)
+	}
+	_, err = b.mgr.SessionForGuild(gID).FollowupMessageCreate(i.Interaction, false, &discordgo.WebhookParams{
+		Content: fmt.Sprintf("Tibia world for this server set to: **%s**", world),
+	})
+	return err
+}
+
+func (b *Bot) ConfigureWorldNameForGuild(guildID string) {
+	guildWorld, err := b.worldNameRepo.SelectGuildWorld(context.Background(), guildID)
+	if err == nil && guildWorld != nil && guildWorld.WorldName != "" {
+		b.onlineCheckService.ConfigureWorldName(guildID, guildWorld.WorldName)
+	}
+}
```

Comment:
The infrastructure layer of Discord should: 

* take user input, 
* sanitize and validate it, 
* and then pass it to a service that implements business logic (in this case - that connects a specific guild to a specific world), 
* then provide feedback (operation result) to the user. 

We shouldn't call database, or implement anything "logic" oriented in the infrastructure layer. We should only take the input, sanitize it, and pass it to a service that handles it, and then forward the result of this operation. 

Let's move this to `onlinecheck` service, e.g. `onlinecheck.(guild, world)` or something like that.